### PR TITLE
Add `ignore: "blockless-group"` option for `at-rule-empty-line-before`

### LINKED ISSUE #1413: Add ignore: "blockless-group" option to at-rule-empty-line-before
Ref: https://github.com/stylelint/stylelint/issues/1377 & https://github.com/stylelint/stylelint/issues/1366

### PR DESCRIPTION
**Pull request by Nirazul**
Closes #1413

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@Nirazul thanks for PR

**Comment by evilebottnawi** [comment]
@jeddy3 LGTM

File: src/rules/at-rule-empty-line-before/__tests__/index.js [hunk 1]
```diff
@@ -121,6 +121,35 @@ testRule(rule, mergeTestDescriptions(sharedAlwaysTests, {
...
+    code: "@media {}; @import 'x.css';",
+  }],
+
+  reject: [{
+    code: "@import 'x.css'; @media {};",
+    message: messages.expected,
+  }],
+})
+
+testRule(rule, {
```
Comment (davidtheclark) [inline src/rules/at-rule-empty-line-before/__tests__/index.js]: No need for these SCSS tests, as they don't contain any syntax that would violate the CSS parser, right? Unless I'm wrong, I'd suggest just adding the `@content;` line to a test in the block above and eliminating the SCSS tests.

File: src/rules/at-rule-empty-line-before/__tests__/index.js [hunk 2]
```diff
@@ -298,6 +327,34 @@ testRule(rule, mergeTestDescriptions(sharedNeverTests, {
...
+    `,
+  } ],
+
+  reject: [{
+    code: `
+      @import 'x.css';
+
+      @media {};
+    `,
+    message: messages.rejected,
```
Comment (davidtheclark) [inline src/rules/at-rule-empty-line-before/__tests__/index.js]: Add line/column so it's clear which at-rule triggered the warning.

**Comment by davidtheclark** [comment]
Thanks @Nirazul! I added a couple of comments, and looks like it will need a rebase.

**Comment by Nirazul** [comment]
I'm gonna close this one, as there has gone something wrong with rebasing 😬 
➡️  #1419

**Pull request closed without merging**